controllers/core/configmap_controller.go

```diff
@@ -67,11 +68,27 @@ func (r *ConfigMapReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 	// Check if the flag value has changed
 	newWinIPAMEnabledCond := r.Condition.IsWindowsIPAMEnabled()
 
+	var isIPAMFlagUpdated bool
 	if r.curWinIPAMEnabledCond != newWinIPAMEnabledCond {
 		r.curWinIPAMEnabledCond = newWinIPAMEnabledCond
 		logger.Info("updated configmap", config.EnableWindowsIPAMKey, r.curWinIPAMEnabledCond)
 
-		// Flag is updated, update all nodes
+		isIPAMFlagUpdated = true
+	}
+
+	// Check if the prefix delegation flag has changed
+	newWinPrefixDelegationEnabledCond := r.Condition.IsWindowsPrefixDelegationEnabled()
+
+	var isPrefixFlagUpdated bool
+	if r.curWinPrefixDelegationEnabledCond != newWinPrefixDelegationEnabledCond {
+		r.curWinPrefixDelegationEnabledCond = newWinPrefixDelegationEnabledCond
+		logger.Info("updated configmap", config.EnableWindowsPrefixDelegationKey, r.curWinPrefixDelegationEnabledCond)
+
+		isPrefixFlagUpdated = true
```

Comment:
Can you help me understand the diff between `isPrefixFlagUpdated` and `r.curWinPrefixDelegationEnabledCond` both which are boolean and referring to the same state of a field in the CM... I am not sure I follow why we need two.

Reply:
Yea sure. Since `r.curWinPrefixDelegationEnabledCond` is being overwritten here with the new value, there is no way to check if this condition is updated outside this `if` block. So I added a boolean flag here so that later on we can check if this value is indeed updated by new values in config map. If the value in config map is not new, then effectively there is no update and hence no further update on nodes.

Reply:
Do you mean you use these two new variables to trigger `UpdateNodesOnConfigMapChanges()`?

Reply:
Yes exactly